src/stores/AppStore.ts

```diff
@@ -325,14 +325,21 @@ export class AppStore {
             }
         }
         this.logStore.addInfo(`Loaded file ${ack.fileInfo.name} with dimensions ${dimensionsString}`, ["file"]);
+        const beamTable: CARTA.Beam[] = [];
+        if(ack.beamTable && ack.beamTable.length > 0) {
+            for (const beam of ack.beamTable) {
+                beamTable.push(new CARTA.Beam(beam));
+            }
+        }
         const frameInfo: FrameInfo = {
             fileId: ack.fileId,
             directory,
             hdu,
             fileInfo: new CARTA.FileInfo(ack.fileInfo),
             fileInfoExtended: new CARTA.FileInfoExtended(ack.fileInfoExtended),
             fileFeatureFlags: ack.fileFeatureFlags,
-            renderMode: CARTA.RenderMode.RASTER
+            renderMode: CARTA.RenderMode.RASTER,
+            beamTable: beamTable
```

Comment:
can't we just use `ack.beamTable ?? []` here? Why do we need to manually clone the beam table?